core/src/main/java/com/netflix/iceberg/util/BinPacking.java

```diff
@@ -107,6 +109,8 @@ public boolean hasNext() {
           bins.addLast(bin);
 
           if (bins.size() > lookback) {
+            // sort by bin weight, descending
+            bins.sort(Comparator.comparing(Bin::getWeight).reversed());
```

Comment:
Modifying the order of the remaining bins changes the behavior of the algorithm more than intended. I think this should remove the bin with the largest weight, but should not reorder the other bins.

That's also going to waste less time because removing the highest weight bin requires scanning all N bins, but sorting them is N*log(N).